configure.ac

```diff
@@ -1444,6 +1450,10 @@ fi
 
 AM_CONDITIONAL([ENABLE_EXTERNAL_SIGNER], [test "x$use_external_signer" = "xyes"])
 
+if test x$use_trivial_auto_var_init_pattern = xyes; then
+  AX_CHECK_COMPILE_FLAG([-ftrivial-auto-var-init=pattern],[CXXFLAGS="$CXXFLAGS -ftrivial-auto-var-init=pattern"],,[[$CXXFLAG_WERROR]])
```

Comment:
```suggestion
if test x$use_trivial_auto_var_init_pattern != xno; then
  AX_CHECK_COMPILE_FLAG([-ftrivial-auto-var-init=$use_trivial_auto_var_init_pattern],[CXXFLAGS="$CXXFLAGS -ftrivial-auto-var-init=$use_trivial_auto_var_init_pattern"],,[[$CXXFLAG_WERROR]])
```

Reply:
I avoided that because the only options are `uninitialized` (already default), `zero` (needs `-enable-trivial-auto-var-init-zero-knowing-it-will-be-removed-from-clang`), `pattern` (can be enabled with the new option).

Happy to apply your patch, but I think it also needs `-enable-trivial-auto-var-init-zero-knowing-it-will-be-removed-from-clang`, which again will throw a warning if used with `pattern`. As this is only used for testing, a warning is fine by me.